doc/distributed-data-plane-verification/design.md

```diff
@@ -0,0 +1,276 @@
+
+# Revision
+
+| Rev   | Date  | Author | Change Description |
+| ---- | ---------- | -----------| ------------------|
+| v0.1 | 02/24/2022 | Qiao Xiang, Chenyang Huang, Ridi Wen, Yuxin Wang, Jiwu Shu@Xiamen University, China| Initial version |
+
+
+# Scope
+This document describes the high-level design of the distributed data plane verification feature.
+
+
+
+
+# Definitions/Abbreviations
+###### Table 1: Abbreviations
+| Abbreviation |          Full form          |
+| ------------ | --------------------------- |
+| FIB          | Forwarding Information Base |
+| CLI          | Command Line Interface      |
+| LEC          | Local Equivalence Class     |
+| DPV          | Data plane verification     |
+
+# Overview
+
+Network errors such as forwarding loops, undesired blackholes and waypoint violations are the outcome of various
+issues (e.g., software bugs, faulty hardware, protocol misconfigurations, and oversight negligence). They can happen in
+all types of networks (e.g., enterprise networks, wide area networks and data center networks), and have both disastrous
+financial and social consequences. Data plane verification (DPV) is important for finding such network errors. 
+Current DPV tools employ a centralized architecture, where a server collects the data planes of all devices and verifies them. 
+Despite substantial efforts on accelerating DPV, this centralized architecture is inherently unscalable because (1) it requires a highly available management network, which is hard build itself; and (2) the server becomes the performance bottleneck and a single point of failure. 
+
+In this HLD, to tackle the scalability challenge of DPV, we propose a distributed
+data plane verification feature, which circumvents the scalability bottleneck
+of centralized design and performs data plane checking on commodity network
+devices.  Our key insight is that DPV can be transformed into a counting problem
+on a directed acyclic graph called DVNet, which can be naturally decomposed into lightweight
+tasks executed at network devices, enabling scalability. To be concrete, this
+feature provides:
+
+* A declarative requirement specification language that allows operators to
+  flexibly specify common requirements studied by existing DPV tools (e.g.,
+reachability, blackhole free, waypoint and isolation), and more advanced, yet
+understudied requirements (e.g., multicast, anycast, no-redundant-delivery and
+all-shortest-path availability).
+* A verification planner that takes the operator specified requirement as input,
+  and systematically decomposes the global verification into lightweight,
+on-device computation tasks.
+* A distributed verification (DV) protocol that specifies how on-device
+  verifiers communicate task results efficiently to collaboratively verify the
+requirements.
+
+
+
+The picture below demonstrates the architecture and workflow of distributed data plane verification.
+
+
+![system](img/system-diagram.jpg)
```

Comment:
The system-diagram.jpg should give more details. For example, what container to add? How do new containers interact with existing components?